Source/DafnyCore/Resolver/CallGraphBuilder.cs

```diff
@@ -266,7 +266,16 @@ private class CallGraphASTVisitor : ASTVisitor<CallGraphBuilderContext> {
         ModuleDefinition callerModule = context.CodeContext.EnclosingModule;
         ModuleDefinition calleeModule = ((IASTVisitorContext)callee).EnclosingModule;
         if (callerModule != calleeModule) {
-          // inter-module call; don't record in call graph
+          // inter-module call; add edge in module's inter-module call graph
+          if (context.CodeContext is ICallable caller0) {
```

Comment:
I tried adding this guard, only to discover that the CallGraphBuilder class does not have access to Dafny options. Moreover, functions of this class are always called statically from outside. 

To implement this extra guard, I will have to add an extra parameter to every callable in this class to store whether to build the interModuleCallGraph. Should I do that? In my opinion that is a bigger change which probably isn't justified by the relatively small savings from not building the intermodule call graph.

Reply:
Since this is an instance function, you should be able to store a Dafny options as a field, so you don't need to change all instance functions.
For static methods, it's ok to put a DafnyOption as an arguments.

I don't know how much time is spent building the inter module call graph but it might be pretty long for codebases with 6000+ methods. If you don't want to do this change, could you please confirm that the extra time of doing that computation for a big project does not exceed 100ms?

Reply:
I think the cost of this computation should be small, since only the functions directly called by a method are added to the intermodule call graph, and each individual method usually has only about 6 to 10 "depth 1" dependencies.

What would be the best way to check the computation overhead for the entire project? Anything more sophisticated than running `time dafny verify ...` on a few files multiple times to get an average, and comparing the time taken on master vs my branch?

Reply:
Okay, I ran it multiple times both with and without the intermodule graph stuff commented out. The runtime numbers were noisy so hard to estimate exactly how much it is costing, but nothing appreciable since about half the time the present code was faster and half the time, the code with it commented out was faster.

Reply:
Thanks! I feel better.